packages/common/test/util/http.test.js

```diff
@@ -215,6 +215,41 @@ describe('request function', () => {
     expect(response.url).to.eql('https://www.example.com/api');
   });
 
+  it('should return undefined if response body is empty and parseAs is json', async () => {
+    client
+      .intercept({
+        path: '/api',
+        method: 'PUT',
+      })
+      .reply(200, undefined, {
+        headers: { 'Content-Length': '0' },
+      });
+
+    const response = await request('PUT', 'https://www.example.com/api', {
+      parseAs: 'json',
+      body: { id: 2 },
+    });
+
+    expect(response.statusCode).to.eql(200);
+    expect(response.body).to.eql(undefined);
+  });
+
+  it('should throw an error if there is no content-length header and an empty response body', async () => {
```

Comment:
This is an EXCELLENT unit test because it demonstrates some really important behaviour in this function.

Looking at this unit test, I'm thinking, "we haven't really got this behaviour right yet". Because if the body is empty we really shouldn't throw an error, no matter what the state of the HTTP headers.

However, I understand WHY we have that behaviour and I don't think I'm going to ask you to change anything. I just want to highlight how useful this test is in explaining and documenting the behaviour of the functoin.